### ISSUE TITLE
allow conditions like `ts < expr(now())` in temporal filter

### ISSUE THREAD

**Issue by stdrc** [risingwavelabs/risingwave#17630]

**Comment by stdrc** [comment]
Actually this is already done after #11454 and #14098.

**Comment by st1page** [comment]
```
dev=> explain create materialized view mv as     select * from t1 where ts + interval '1 hour' < date_trunc('DAY',now());
ERROR:  Failed to run the query

Caused by:
  Conditions containing now must be of the form `input_expr cmp now() [+- const_expr]` or `now() [+- const_expr] cmp input_expr`, where `input_expr` references a column and contains no `now()`.
```
